cmd/maya-apiserver/spc-watcher/lease.go

```diff
@@ -86,18 +177,79 @@ func (sl *spcLease) UpdateLease(leaseValue string) (*apis.StoragePoolClaim, erro
 	return spcObject, nil
 }
 
-// Can be used for reconcile loop use cases
-// TODO Remove using patch instead of update
-func (sl *spcLease) RemoveLease() *apis.StoragePoolClaim {
-	spcObject, err := sl.UpdateLease("")
+func (sl *spcLease) RemoveLease() {
+	_, err := sl.patchSpc()
 	if err != nil {
-		runtime.HandleError(fmt.Errorf("Lease could not be removed:%v", err))
+		newErr := fmt.Errorf("Lease could not be removed:%v", err)
+		runtime.HandleError(newErr)
 	}
-	return spcObject
+	glog.Info("Lease removed successfully on storagepoolclaim")
 }
 
 func (sl *spcLease) getPodName() string {
-	podName := menv.Get(menv.OpenEBSMayaPodName)
-	podNameSpace := menv.Get(menv.OpenEBSNamespace)
+	podName := env.Get(env.OpenEBSMayaPodName)
+	podNameSpace := env.Get(env.OpenEBSNamespace)
 	return podNameSpace + "/" + podName
 }
+
+// patchSpc will patch the spc object to release the lease
+func (sl *spcLease) patchSpc() (*apis.StoragePoolClaim, error) {
+	spcPatch := make([]SPCPatch, 1)
+	// setting operation as remove
+	spcPatch[0].Op = PatchOperation
+	// object to be removed is finalizers
+	spcPatch[0].Path = PatchPath
+	leaseValueObj, err := parseLeaseValue(sl.spcObject.Annotations[SpcLeaseKey])
+	leaseValueObj.HolderIdentity = ""
+	newLeaseValue, err := json.Marshal(leaseValueObj)
+	if err != nil {
+		return nil, err
+	}
+	spcPatch[0].Value = string(newLeaseValue)
+	spcPatchJSON, err := json.Marshal(spcPatch)
```

Comment:
This is as good as a patch.Provide(). Move this stuff to Patcher interface.

NOTE: If Provide() does not fit, then check if Patcher.Value() can be an alternative.